libafl/src/feedbacks/mod.rs

```diff
@@ -54,49 +47,51 @@ pub mod new_hash_feedback;
 pub mod stdio;
 pub mod transferred;
 
-/// Feedbacks evaluate the observers.
-/// Basically, they reduce the information provided by an observer to a value,
-/// indicating the "interestingness" of the last run.
-pub trait Feedback<S>: Named
-where
-    S: State,
-{
+/// Feedback which initializes a state.
+///
+/// This trait is separate from the general [`Feedback`] definition as it would not be sufficiently
+/// specified otherwise.
+pub trait StateInitializer<S> {
     /// Initializes the feedback state.
     /// This method is called after that the `State` is created.
     fn init_state(&mut self, _state: &mut S) -> Result<(), Error> {
         Ok(())
     }
+}
 
+/// Feedbacks evaluate the observers.
+/// Basically, they reduce the information provided by an observer to a value,
+/// indicating the "interestingness" of the last run.
+pub trait Feedback<EM, I, OT, S>: StateInitializer<S> + Named {
```

Comment:
Do all feedbacks _need_ to be StateInitializers? It feels like only a subset has to

Reply:
Sadly, yes, because all feedbacks get passed to the state. Otherwise, we cannot actually initialize the feedbacks that need this, since we use CombinedFeedback.

Reply:
And it cannot be in Feedback directly? Sad but oh well